sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/models/AmqpDataBody.java

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.amqp.models;
+
+import com.azure.core.util.IterableStream;
+
+import java.util.Objects;
+
+/**
+ * This is amqp message body which represents {@link AmqpBodyType#DATA} type.
+ */
+public final class AmqpDataBody implements AmqpMessageBody {
```

Comment:
I didn't mean to add to the public surface area so it fits our scenario. azure-core-amqp is generic to AMQP protocol. The support for a single BinaryData is an implementation detail because we use proton-j.

Reply:
Okay moving it back to SBM.